src/advanced/mod.rs

```diff
@@ -0,0 +1,123 @@
+//! Niche and advanced, you can ignore this to begin with.
+//!
+//! Module for other [`ForLt!`]-like constructs, or for
+//! [`ForLt`]-adjacent definitions, such as [`WithLifetime`].
+//!
+//! [`ForLt`]: trait@crate::ForLt
+//! [`ForLt!`]: crate::ForLt!
+
+use super::*;
+
+pub
+mod extra_arities;
+
+pub
+mod type_eq;
+
+/// The [`::nougat`](https://docs.rs/nougat) of this crate, which allows "expressing `dyn`-safe
+/// <code>impl [ForLt]</code> types", sort to speak.
+///
+/// The idea is that in order for the <code>: [ForLt]</code> design to be ergonomic,
+/// it is paramount to have a [`ForLt!`]-like companion construct, in the same fashion that in order
+/// for the stdlib [`Fn…`][FnOnce] traits to be ergonomic and useful, it is paramount to have
+/// literal `|…| { … }` closures expressions as the dual construct.
+///
+/// And the trick to achieve producing an on-the-fly `impl ForLt` type, _i.e._, an on-the-fly
+/// type which associates, `for</* any */ 'lifetime>`, a type to such `'lifetime`, is to involve a
+/// `dyn for<'any_lt> SomeTrait<'any_lt, Assoc = …>` of some sorts.
+///
+/// [`WithLifetime`] is such a `SomeTrait`.
+///
+/// Actual usefulness of this trait for downstream users is to be seen, but since there seemed to be
+/// some interest, let's expose it and see how users end up using it.
+///
+/// [ForLt]: trait@ForLt
+pub
+trait WithLifetime<'lt>
```

Comment:
I would suggest sealing this trait too; this would ensure that for all `T: ForLt`, `ForLt![T::Of<'_>] == T` (and more generally, it preserves the bijection between `ForLt` types and `for<'a> $ty<'a>`  "HKTs").

This is mostly useful when implementing lifetime-infected downcasting using `TypeId::of<T: ForLt + 'static>()`, as it ensures that `T` and `ForLt![T::Of<'_>]` always have the same type id.

Reply:
Leaving `WithLifetime` unsealed would allow users to manually implement it, but I'm not sure how useful it is:
```rust
// Is this:
pub type MyFamily = ForLt![Some<'_, Complex<Type<'_>, u32>];

// Always better than this?
pub struct MyFamily;
impl<'a> WithLifetime<'a> {
  type Of = Some<'a, Complex<Type<'a>, u32>
}
```